log/uds.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"go.uber.org/zap/zapcore"
+)
+
+// An udsCore write entries to an UDS server
+type udsCore struct {
+	client       http.Client
+	minimumLevel zapcore.Level
+	url          string
+	maxAttempts  int
+}
+
+// teeToUDSServer returns a zapcore.Core that writes entries to both the provided core and to an uds server.
+func teeToUDSServer(baseCore zapcore.Core, address, path string, maxRetryAttempts int) zapcore.Core {
+	c := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", address)
+			},
+		},
+		Timeout: 100 * time.Millisecond,
+	}
+	uc := &udsCore{client: c, url: "http://unix" + path, maxAttempts: maxRetryAttempts + 1}
+	for l := zapcore.DebugLevel; l <= zapcore.FatalLevel; l++ {
+		if baseCore.Enabled(l) {
+			uc.minimumLevel = l
```

Comment:
isn't this per-scope?

Reply:
Scope is one level above. The minimumLevel here is always debug level if I understand correctly.